runtime/mainnet/src/lib.rs

```diff
@@ -1040,4 +1132,198 @@ mod tests {
+	}
+
+	const ALICE: [u8; 32] = [1; 32];
+	pub fn new_test_ext() -> sp_io::TestExternalities {
+		let initial_balance = 100_000_000 * UNIT;
+		let mut t = frame_system::GenesisConfig::<Runtime>::default().build_storage().unwrap();
+		pallet_balances::GenesisConfig::<Runtime> {
+			balances: vec![
+				(TreasuryAccount::get(), initial_balance),
+				(MaintenanceAccount::get(), initial_balance),
+				(ALICE.into(), initial_balance),
+			],
+		}
+		.assimilate_storage(&mut t)
+		.unwrap();
+		let mut ext = sp_io::TestExternalities::new(t);
+		ext.execute_with(|| System::set_block_number(1));
+		ext
+	}
+
+	#[test]
+	fn transaction_payment_charges_fees_via_balances_and_funds_treasury_and_maintenance_equally() {
+		new_test_ext().execute_with(|| {
+			let who: AccountId = ALICE.into();
+			let call = RuntimeCall::System(frame_system::Call::remark { remark: vec![] });
+			let fee = UNIT / 10;
+			let tip = UNIT / 2;
+			let fee_plus_tip = fee + tip;
+			let treasury_balance =
+				pallet_balances::Pallet::<Runtime>::free_balance(&TreasuryAccount::get());
+			let maintenance_balance =
+				pallet_balances::Pallet::<Runtime>::free_balance(&MaintenanceAccount::get());
+			let who_balance = pallet_balances::Pallet::<Runtime>::free_balance(&who);
+			let dispatch_info = call.get_dispatch_info();
+
+			// NOTE: OnChargeTransaction functions expect tip to be included within fee
+			let liquidity_info =
+				<OnChargeTransaction as OnChargeTransactionT<Runtime>>::withdraw_fee(
+					&who,
+					&call,
+					&dispatch_info,
+					fee_plus_tip,
+					0,
+				)
+				.unwrap();
+			<OnChargeTransaction as OnChargeTransactionT<Runtime>>::correct_and_deposit_fee(
+				&who,
+				&dispatch_info,
+				&call.dispatch(RuntimeOrigin::signed(who.clone())).unwrap(),
+				fee_plus_tip,
+				0,
+				liquidity_info,
+			)
+			.unwrap();
+
+			let treasury_expected_balance = treasury_balance + (fee_plus_tip / 2);
+			let maintenance_expected_balance = maintenance_balance + (fee_plus_tip / 2);
+			let who_expected_balance = who_balance - fee_plus_tip;
+
+			assert!(treasury_balance != 0);
+			assert!(maintenance_expected_balance != 0);
+
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&TreasuryAccount::get()),
+				treasury_expected_balance
+			);
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&MaintenanceAccount::get()),
+				maintenance_expected_balance
+			);
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&who),
+				who_expected_balance
+			);
+		})
+	}
+
+	#[test]
+	fn test_fees_and_tip_split() {
+		new_test_ext().execute_with(|| {
+			let fee_amount = 10;
+			let fee =
+				<pallet_balances::Pallet<Runtime> as frame_support::traits::fungible::Balanced<
+					AccountId,
+				>>::issue(fee_amount);
+			let tip_amount = 20;
+			let tip =
+				<pallet_balances::Pallet<Runtime> as frame_support::traits::fungible::Balanced<
+					AccountId,
+				>>::issue(tip_amount);
```

Comment:
```suggestion
				<pallet_balances::Pallet<Runtime> as fungible::Balanced<
					AccountId,
				>>::issue(fee_amount);
			let tip_amount = 20;
			let tip =
				<pallet_balances::Pallet<Runtime> as fungible::Balanced<
					AccountId,
				>>::issue(tip_amount);
```

non-blocking, but unnecessary path prefixes (already imported).

Reply:
https://github.com/r0gue-io/pop-node/pull/428/commits/5b6f7315528a1dfd987eb5bfa2abd01949e8cf5a